dpytools/validation/json/validation.py

```diff
@@ -0,0 +1,86 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+from urllib.parse import urlparse
+
+import jsonschema
+from jsonschema import ValidationError
+
+
+def validate_json_schema(
+    schema_path: Union[Path, str],
+    data_dict: Optional[Dict] = None,
+    data_path: Union[Path, str, None] = None,
+    error_msg: Optional[str] = "",
+    indent: Optional[int] = 2,
+):
+    """
+    Validate metadata.json files against schema.
+
+    Either `data_dict` or `data_path` must be provided.
+
+    `error_msg` and `indent` are used to format the error message if validation fails.
+    """
+    # Confirm that *either* `data_dict` *or* `data_path` has been provided, otherwise raise ValueError
+    if data_dict and data_path:
+        raise ValueError(
+            "Both a dictionary and file path of data have been provided - please specify either one or the other, not both."
+        )
+    if data_dict is None and data_path is None:
+        raise ValueError(
+            "Please provide either a dictionary or a file path of the data to be validated against the schema."
+        )
+
+    # Load schema as dict
+    if isinstance(schema_path, str):
+        parsed_schema_path = urlparse(schema_path)
+        if parsed_schema_path.scheme == "http":
+            # TODO Load schema from URL
+            raise NotImplementedError("Validation from remote schema not yet supported")
+        # Convert `schema_path` to pathlib.Path
+        schema_path = Path(schema_path).absolute()
+    # Check `schema_path` exists
+    if not schema_path.exists():
+        raise ValueError(f"Schema path '{schema_path}' does not exist")
+    with open(schema_path, "r") as f:
+        schema_from_path = json.load(f)
+
+    # Load data to be validated
+    if data_dict:
+        if not isinstance(data_dict, Dict):
+            raise ValueError("Invalid data format")
```

Comment:
"Invalid data format, data_dict should be a python dictionary". Let's really make it obvious.